pyomo/contrib/parmest/tests/test_parmest_cov.py

```diff
@@ -0,0 +1,2590 @@
+                theta: vector of fitted parameters
+                t: independent variable [hours]
+                y: dependent variable [?]
+            """
+            return y - model(theta, t)
+
+        # define data
+        t = self.data["hour"].to_numpy()
+        y = self.data["y"].to_numpy()
+
+        # define initial guess
+        theta_guess = np.array([15, 0.5])
+
+        ## solve with optimize.least_squares
+        sol = scipy.optimize.least_squares(
+            residual, theta_guess, method="trf", args=(t, y), verbose=2
+        )
+        theta_hat = sol.x
+
+        self.assertAlmostEqual(
+            theta_hat[0], 19.1426, places=2
+        )  # 19.1426 from the paper
+        self.assertAlmostEqual(theta_hat[1], 0.5311, places=2)  # 0.5311 from the paper
+
+        # calculate residuals
+        r = residual(theta_hat, t, y)
+
+        # calculate variance of the residuals
+        # -2 because there are 2 fitted parameters
+        sigre = np.matmul(r.T, r / (len(y) - 2))
+
+        # approximate covariance
+        # Need to divide by 2 because optimize.least_squares scaled the objective by 1/2
+        cov = sigre * np.linalg.inv(np.matmul(sol.jac.T, sol.jac))
+
+        self.assertAlmostEqual(cov[0, 0], 6.22864, places=2)  # 6.22864 from paper
+        self.assertAlmostEqual(cov[0, 1], -0.4322, places=2)  # -0.4322 from paper
+        self.assertAlmostEqual(cov[1, 0], -0.4322, places=2)  # -0.4322 from paper
+        self.assertAlmostEqual(cov[1, 1], 0.04124, places=2)  # 0.04124 from paper
+
+    def test_cov_scipy_curve_fit_comparison(self):
+        """
+        Scipy results differ in the 3rd decimal place from the paper. It is possible
+        the paper used an alternative finite difference approximation for the Jacobian.
+        """
+
+        ## solve with optimize.curve_fit
+        def model(t, asymptote, rate_constant):
+            return asymptote * (1 - np.exp(-rate_constant * t))
+
+        # define data
+        t = self.data["hour"].to_numpy()
+        y = self.data["y"].to_numpy()
+
+        # define initial guess
+        theta_guess = np.array([15, 0.5])
+
+        theta_hat, cov = scipy.optimize.curve_fit(model, t, y, p0=theta_guess)
+
+        self.assertAlmostEqual(
+            theta_hat[0], 19.1426, places=2
+        )  # 19.1426 from the paper
+        self.assertAlmostEqual(theta_hat[1], 0.5311, places=2)  # 0.5311 from the paper
+
+        self.assertAlmostEqual(cov[0, 0], 6.22864, places=2)  # 6.22864 from paper
+        self.assertAlmostEqual(cov[0, 1], -0.4322, places=2)  # -0.4322 from paper
+        self.assertAlmostEqual(cov[1, 0], -0.4322, places=2)  # -0.4322 from paper
+        self.assertAlmostEqual(cov[1, 1], 0.04124, places=2)  # 0.04124 from paper
+
+# Test class for when the user wants to use the built-in Parmest SSE_weighted objective function
+@unittest.skipIf(not ipopt_available, "The 'ipopt' command is not available")
+class TestRooneyBieglerWSSE(unittest.TestCase):
```

Comment:
@slilonfe5 Do these tests give the same answers for WSSE and SEE? If so, we should talk to the Pyomo team about how to reduce the amount of redundant testing code.